### PR TITLE
Upgrade to 2.5.0

### PR DESCRIPTION
**Pull request by metinavci**

### CODE REVIEW

File: SPIClient/Cashout.cs [hunk 1]
```diff
@@ -31,7 +28,6 @@ public Message ToMessage()
                 new JProperty("surcharge_amount", SurchargeAmount)
                 );
             Config.addReceiptConfig(data);
-            Options.AddOptions(data);
```
Comment (metinavci) [inline SPIClient/Cashout.cs]: Receipt Footer and Header setting shouldn't be in Cashout Transaction.

File: SPIClient/Purchase.cs [hunk 2]
```diff
@@ -453,7 +451,6 @@ public Message ToMessage()
                 new JProperty("suppress_merchant_password", IsSuppressMerchantPassword)
             );
             Config.addReceiptConfig(data);
-            Options.AddOptions(data);
```
Comment (metinavci) [inline SPIClient/Purchase.cs]: Receipt Footer and Header setting shouldn't be in Refund Transaction

File: SPIClient/Settlement.cs [hunk 3]
```diff
@@ -14,14 +14,19 @@ public class SettleRequest
...
         {
             Id = id;
         }
-        
+
         public Message ToMessage()
         {
-            return new Message(Id, Events.SettleRequest, null, true);
+            var data = new JObject();
+            Config.addReceiptConfig(data);
```
Comment (RaymondK-AP) [inline SPIClient/Settlement.cs]: Can you please change the casing for the function 'addReceiptConfig' to 'AddReceiptConfig'

File: SPIClient/Purchase.cs [hunk 4]
```diff
@@ -661,28 +661,26 @@ public class MotoPurchaseRequest
...
         {
             var data = new JObject(
                 new JProperty("pos_ref_id", PosRefId),
                 new JProperty("purchase_amount", PurchaseAmount),
-                new JProperty("surcharge_amount", SurchargeAmount)
+                new JProperty("surcharge_amount", SurchargeAmount),
+                new JProperty("suppress_merchant_password", IsSuppressMerchantPassword)
             );
             Config.addReceiptConfig(data);
-            Options.AddOptions(data);
```
Comment (metinavci) [inline SPIClient/Purchase.cs]: Receipt Footer and Header shouldn't be in Moto Purchase Transaction

File: SPIClient/Spi.cs [hunk 5]
```diff
@@ -790,12 +795,18 @@ public InitiateTxResult InitiateSettleTx(string posRefId)
             lock (_txLock)
             {
                 if (CurrentFlow != SpiFlow.Idle) return new InitiateTxResult(false, "Not Idle");
-                var settleRequestMsg = new SettleRequest(RequestIdHelper.Id("settle")).ToMessage();
+                var settleRequest = new SettleRequest(RequestIdHelper.Id("settle"))
+                {
+                    Config = Config
+                };
+
+                var settleMessage = settleRequest.ToMessage();
```
Comment (RaymondK-AP) [inline SPIClient/Spi.cs]: You can combine this into one, and leave the variable name as settleRequestMsg

File: SPIClient/Settlement.cs (hunk 3 above)
Comment (metinavci) [reply SPIClient/Settlement.cs]: Done

File: SPIClient/Spi.cs (hunk 5 above)
Comment (metinavci) [reply SPIClient/Spi.cs]: Done

File: SPIClient/Spi.cs [hunk 6]
```diff
@@ -814,12 +810,10 @@ public InitiateTxResult InitiateSettleTx(string posRefId)
             lock (_txLock)
             {
                 if (CurrentFlow != SpiFlow.Idle) return new InitiateTxResult(false, "Not Idle");
-                var settleRequest = new SettleRequest(RequestIdHelper.Id("settle"))
+                var settleMessage = new SettleRequest(RequestIdHelper.Id("settle"))
```
Comment (RaymondK-AP) [inline SPIClient/Spi.cs]: can you please rename to settleMsg

Comment (metinavci) [reply SPIClient/Spi.cs]: it's done

File: SPIClient/Purchase.cs [hunk 7]
```diff
@@ -661,28 +661,26 @@ public class MotoPurchaseRequest
     {
         public string PosRefId { get; }
         public int PurchaseAmount { get; }
-        public int SurchargeAmount { get; }
+        public int SurchargeAmount { get; set; }
+        public bool IsSuppressMerchantPassword { get; set; }
```
Comment (RaymondK-AP) [inline SPIClient/Purchase.cs]: Can you please rename the variable to SuppressMerchantPassword thanks

File: SPIClient/Spi.cs [hunk 8]
```diff
@@ -762,15 +762,34 @@ public InitiateTxResult InitiateMotoPurchaseTx(string posRefId, int amountCents)
...
+
+        /// <summary>
+        /// Initiates a Mail Order / Telephone Order Purchase Transaction
+        /// </summary>
+        /// <param name="posRefId">Alphanumeric Identifier for your transaction.</param>
+        /// <param name="amountCents">Amount in Cents</param>
+        /// <param name="surchargeAmount">The Surcharge Amount in Cents</param>
+        /// <param name="isSuppressMerchantPassword">>Merchant Password control in VAA</param>
+        /// <returns>InitiateTxResult</returns>
+        public InitiateTxResult InitiateMotoPurchaseTx(string posRefId, int amountCents, int surchargeAmount, bool isSuppressMerchantPassword)
```
Comment (RaymondK-AP) [inline SPIClient/Spi.cs]: Same, rename to suppressMerchantPassword

File: SPIClient/Purchase.cs (hunk 7 above)
Comment (metinavci) [reply SPIClient/Purchase.cs]: Done

File: SPIClient/Spi.cs (hunk 8 above)
Comment (metinavci) [reply SPIClient/Spi.cs]: Done

**Pull request merged**